order/src/main/java/com/yas/order/service/OrderService.java

```diff
@@ -198,4 +203,26 @@ public List<OrderGetVm> getMyOrders(String productName, EOrderStatus orderStatus
         List<Order> orders = orderRepository.findMyOrders(userId, productName, orderStatus);
         return orders.stream().map(OrderGetVm::fromModel).toList();
     }
+
+    public void exportCSVFile(HttpServletResponse response, List<Long> orderIdList) {
+        //set file name and content type
+        String filename = "Orders-data.csv";
+
+        response.setContentType("text/csv");
+        response.setHeader(HttpHeaders.CONTENT_DISPOSITION,
+                "attachment; filename=\"" + filename + "\"");
+        response.setCharacterEncoding("UTF-8");
+        //create a csv writer
+        try {
+            StatefulBeanToCsv<OrderCsvExportVm> writer = new StatefulBeanToCsvBuilder<OrderCsvExportVm>(response.getWriter())
+                    .withQuotechar(CSVWriter.NO_QUOTE_CHARACTER)
+                    .withSeparator(CSVWriter.DEFAULT_SEPARATOR)
+                    .withOrderedResults(true)
+                    .build();
+            //write all orders data to csv file
+            writer.write(orderRepository.findByIdIn(orderIdList).stream().map(OrderCsvExportVm::fromModel).toList());
+        } catch (Exception ex) {
+            throw new InternalException(ex.getMessage());
```

Comment:
We might lost the stack trace of the origin Exception here, can we remove the try/catch here or throw exception more properly

Reply:
I'll alter it when i'm available, Thanks for your comment.